include/hal/common-mock-func.sh

```diff
@@ -0,0 +1,496 @@
+TEST_FLASH_LOCK="${TEST_FLASH_LOCK:-}"
+TEST_BOARD_HAS_FD_REGION="${TEST_BOARD_HAS_FD_REGION:-true}"
+TEST_BOARD_FD_REGION_RW="${TEST_BOARD_FD_REGION_RW:-true}"
+TEST_BOARD_HAS_ME_REGION="${TEST_BOARD_HAS_ME_REGION:-true}"
+TEST_BOARD_ME_REGION_RW="${TEST_BOARD_ME_REGION_RW:-true}"
+TEST_BOARD_ME_REGION_LOCKED="${TEST_BOARD_ME_REGION_LOCKED:-}"
+TEST_BOARD_HAS_GBE_REGION="${TEST_BOARD_HAS_GBE_REGION:-true}"
+TEST_BOARD_GBE_REGION_RW="${TEST_BOARD_GBE_REGION_RW:-true}"
+TEST_BOARD_GBE_REGION_LOCKED="${TEST_BOARD_GBE_REGION_LOCKED:-}"
+TEST_COMPATIBLE_EC_VERSINO="${TEST_COMPATIBLE_EC_VERSINO:-}"
+# This is 2*1024*1024:
+TEST_FLASH_CHIP_SIZE="${TEST_FLASH_CHIP_SIZE:-2097152}"
+
+flashrom_check_flash_lock_mock(){
+# For flash lock testing, for more inf. check check_flash_lock func.:
+  if [ "$TEST_FLASH_LOCK" = "true" ]; then
+    echo "PR0: Warning:.TEST is read-only\|SMM protection is enabled" > /tmp/check_flash_lock.err
+    return 1
+  fi
+
+  return 0
+}
+
+flashrom_flash_chip_name_mock(){
+# For flash chip name check emulation, for more inf. check check_flash_chip
+# func.:
+    echo "Test Flash Chip"
+
+    return 0
+}
+
+flashrom_flash_chip_size_mock(){
+# For flash chip size check emulation, for more inf. check check_flash_chip
+# func..
+  echo "$TEST_FLASH_CHIP_SIZE"
+
+  return 0
+}
+
+flashrom_check_intel_regions_mock(){
+# For flash regions check emulation, for more inf. check check_intel_regions
+# func.:
+  [ "$TEST_BOARD_HAS_FD_REGION" = "true" ] && echo "Flash Descriptor region"
+  [ "$TEST_BOARD_FD_REGION_RW" = "true" ] && echo "Flash Descriptor region.testread-write"
+
+  [ "$TEST_BOARD_HAS_ME_REGION" = "true" ] && echo "Management Engine region"
+  [ "$TEST_BOARD_ME_REGION_RW" = "true" ] && echo "Management Engine region.testread-write"
+  [ "$TEST_BOARD_ME_REGION_LOCKED" = "true" ] && echo "Management Engine region.testlocked"
+
+  [ "$TEST_BOARD_HAS_GBE_REGION" = "true" ] && echo "Gigabit Ethernet region"
+  [ "$TEST_BOARD_GBE_REGION_RW" = "true" ] && echo "Gigabit Ethernet region.testread-write"
+  [ "$TEST_BOARD_GBE_REGION_LOCKED" = "true" ] && echo "Gigabit Ethernet region.testlocked"
+
+  return 0
+}
+
+flashrom_read_flash_layout_mock(){
+# For checking flash layout for further flashrom arguments selection, for more
+# inf. check set_flashrom_update_params function.
+#
+# TODO: this one can be deleted in future and replaced with read_firm_mock,
+# which will create a binary with needed bytes appropriately set.
+  # For -r check flashrom man page:
+  local _file_to_write_into
+  # It won't work with "$*", so make shellcheck ignore it:
+  # shellcheck disable=SC2048
+  _file_to_write_into=$(parse_for_arg_return_next "-r" $*)
+  echo "Testing..." > "$_file_to_write_into"
+
+  return 0
+}
+
+flashrom_read_firm_mock(){
+# Emulating dumping of the firmware the platform currently uses. Currently it is
+# writing into text file, that should be changed to binary instead (TODO).
```

Comment:
I was thinking about adding this file via `dd`, setting needed addresses to needed values, but your propositions are good too.

For now, [I modified it so it will not overwrirte files in case user put its on file there](https://github.com/Dasharo/dts-scripts/compare/e748126a96325d132585966533a906be7dcd2597..a29856f198488767178cbe8e15853f96def1fb8b)